dspace-api/src/main/java/org/dspace/app/bulkaccesscontrol/BulkAccessControl.java

```diff
@@ -552,12 +571,96 @@ private void updateBitstreamPolicies(Bitstream bitstream, Item item, BulkAccessC
+            if (resPolicies.isEmpty()) {
+                return;
+            }
+
+            String resPoliciesStr = resPolicies.stream()
+                    .filter(rp -> rp.getAction() == Constants.READ)  // Filter out non-READ actions
+                    .map(rp -> {
+                        return "[" + rp.getRpName() + ", "
+                                + rp.getRpType() + ", "
+                                + rp.getAction() + ", "
+                                + (rp.getEPerson() == null ? null : rp.getEPerson().getEmail()) + ", "
+                                + (rp.getGroup() == null ? null : rp.getGroup().getName()) + ", "
+                                + (rp.getStartDate() == null ? null : rp.getStartDate().toString()) + ", "
+                                + (rp.getEndDate() == null ? null : rp.getEndDate().toString()) + "]";
+                    })
+                    .collect(Collectors.joining(";"));
+
             resourcePolicyService.removePolicies(context, dso, type, Constants.READ);
+
+            if (dso.getType() != Constants.BITSTREAM && dso.getType() != Constants.ITEM) {
+                return;
+            }
+
+            InstallItemService installItemService = ContentServiceFactory.getInstance().getInstallItemService();
+
+            String timestamp = DCDate.getCurrent().toString();
+            EPerson e = context.getCurrentUser();
+
+            if (dso.getType() == Constants.ITEM) {
+                // Add suitable provenance - includes old resource policies, item, action, user, date +
+                // bitstream checksums
+                Item item = (Item) dso;
+
+                // Build some provenance data while we're at it.
+                StringBuilder prov = new StringBuilder();
+                String msg = "Resource policies (" + (Objects.equals(resPoliciesStr, "") ? "empty" : resPoliciesStr)
+                        + ") of item (" + item.getID() + ") were removed";
+                addProvenanceMetadata(context, item, msg);
+            }
+
+            if (dso.getType() == Constants.BITSTREAM) {
+                // Add suitable provenance - includes old resource policices, bitstream, action, user, date +
+                // bitstream checksums
+                Bitstream bitstream = (Bitstream) dso;
+                ClarinItemService clarinItemService = ClarinServiceFactory.getInstance().getClarinItemService();
+                List<Item> items = clarinItemService.findByBitstreamUUID(context, dso.getID());
+                if (items.isEmpty()) {
+                    return;
+                }
+                Item item = items.get(0);
+                // Build some provenance data while we're at it.
+                String msg = "Resource policies (" + (Objects.equals(resPoliciesStr, "") ? "empty" : resPoliciesStr)
+                        + ") of bitstream (" + bitstream.getID() + ") were removed";
```

Comment:
were removed? I see only addProvenanceMetadata, not remove method

Reply:
we are in the removeReadPolicies method